syntax = "proto2";

package proto;

message LogContent {
    required string key = 1;
    required string value = 2;
}

message LogContentRaw {
    required string key = 1;
    required bytes value = 2;
}

message Log {
    required uint32 time = 1;
    repeated LogContent contents = 2;
    optional fixed32 time_ns = 4;
}

message LogRaw {
    required uint32 time = 1;
    repeated LogContentRaw contents = 2;
    optional fixed32 time_ns = 4;
}

message LogTag {
    required string key = 1;
    required string value = 2;
}

message LogGroup {
    repeated Log logs = 1;
    optional string topic = 3;
    optional string source = 4;
    repeated LogTag log_tags = 6;
}

message LogGroupRaw {
    repeated LogRaw logs = 1;
    optional string topic = 3;
    optional string source = 4;
    repeated LogTag log_tags = 6;
}

message LogGroupList
{
    repeated LogGroup log_groups = 1;
}

message LogGroupListRaw
{
    repeated LogGroupRaw log_groups = 1;
}

